resources/simulation.py

```diff
@@ -46,12 +46,14 @@ def next_epoch(self):
                 if ant.try_drop_food(colony):
                     ant.drop_food(colony)
 
-                future_position = ant.move()
+                pheromone_direction = self.find_pheromone_trace(ant.coordinates, ant.pheromone_status, colony.pheromone.pheromone_array, colony, ant.search_radius)
+                future_position = ant.move(pheromone_direction=pheromone_direction)
                 adjusted_position = self.check_future_position(future_position)
                 ant.coordinates = adjusted_position
                 
                 idx_row, idx_col = self.map_ant_coordinates_to_pheromone_index(ant_coordinates = ant.coordinates,
                                                                                colony = colony)
+                colony.pheromone.reduce_pheromones(0.9999, 0.0001)
```

Comment:
This should take place after all ants (of the colony) have left their pheromone.
```suggestion
                colony.pheromone.leave_pheromone(pos = (idx_row, idx_col), pheromone_status = ant.pheromone_status)
           colony.pheromone.reduce_pheromones(0.9999, 0.0001)
```